src/fromager/sources.py

```diff
@@ -270,7 +270,10 @@ def unpack_source(
     return (next(iter(unpack_dir.glob("*"))), True)
 
 
-def patch_source(ctx: context.WorkContext, source_root_dir: pathlib.Path) -> None:
+def patch_source(
```

Comment:
This function is invoked by some of our build overrides, and this change is backwards incompatible.

Could you make the version argument optional and only perform the check if a version is given? That would ensure we can update safely.